src/languageService/kustoLanguageService.ts

```diff
@@ -431,56 +431,37 @@ class KustoLanguageService implements LanguageService {
-                    const commandSelectionEnd = Math.min(rangeEndOffset, command.absoluteEnd) - command.absoluteStart;
-
-                    command.text =
-                        commandSelectionStart > commandSelectionEnd
-                            ? ''
-                            : command.text.substring(commandSelectionStart, commandSelectionEnd);
-
-                    return command;
-                })
-                .filter((command) => command.text.trim() != '');
-
-            const formattedCommands: string[] = commandsInRange.map((command) =>
-                Kusto.Data.Common.CslQueryParser.PrettifyQuery(command.text, '')
-            );
-            const formattedText: string = formattedCommands.join('\r\n\r\n');
+        const commands = this.getFormattedCommandsInDocumentV2(document, rangeStartOffset, rangeEndOffset);
 
-            return [ls.TextEdit.replace(range, formattedText)];
-        });
+        if (!commands.rangeBeforeFormat || commands.formattedCommands.length === 0) {
+            return Promise.as([]);
+        }
 
-        return commands;
+        return Promise.as([ls.TextEdit.replace(commands.rangeBeforeFormat, commands.formattedCommands.join(''))]);
     }
 
     doDocumentformat(document: ls.TextDocument): Promise<ls.TextEdit[]> {
-        const commands = this.getCommandsInDocument(document).then((commands) => {
-            const formattedCommands = commands.map((command) =>
-                Kusto.Data.Common.CslQueryParser.PrettifyQuery(command.text, '')
-            );
-            const formattedDocument = formattedCommands.join('\r\n\r\n');
+        const startPos = document.positionAt(0);
+        const endPos = document.positionAt(document.getText().length);
+        const fullDocRange = ls.Range.create(startPos, endPos);
 
-            const startPos = document.positionAt(0);
-            const endPos = document.positionAt(document.getText().length);
-            const fullDocRange = ls.Range.create(startPos, endPos);
-            return [ls.TextEdit.replace(fullDocRange, formattedDocument)];
-        });
+        const formattedDoc = this.getFormattedCommandsInDocumentV2(document).formattedCommands.join('');
 
-        return commands;
+        return Promise.as([ls.TextEdit.replace(fullDocRange, formattedDoc)]);
     }
 
     doCurrentCommandFormat(document: ls.TextDocument, caretPosition: ls.Position): Promise<ls.TextEdit[]> {
-        const command: k.CslCommand = this.getCurrentCommand(document, document.offsetAt(caretPosition));
-        const start = document.positionAt(command.AbsoluteStart);
-        const end = document.positionAt(command.AbsoluteEnd);
-        const range = ls.Range.create(start, end);
-        return this.doRangeFormat(document, range);
+        const positionResponse = { v: 0 };
+        const script = this.parseDocumentV2(document);
+        if (script.TryGetTextPosition(caretPosition.line, caretPosition.character, positionResponse)) {
```

Comment:
I think it would be better to use  the standard monaco document.positionAt that we're using in other areas in the code